orttraining/orttraining/python/ort_trainer.py

```diff
@@ -648,6 +648,10 @@ def _init_session(self):
                 *self.model_desc_.outputs_,
                 IODescription(get_all_gradients_finite_arg_name(self.session), [1], torch.bool)]
 
+        if self.state_dict_:
+            self.load_state_dict(self.state_dict_, self.strict_) 
```

Comment:
@liqunfu Could you add some comments here explaining how this works and why it is needed? thanks.

Reply:
It looks like .. if the ONNX model has not been initialized and load_state_dict(...) is called, then ORTrainer(...) only stores an internal reference to desired state. Later when sample inputs are provided and the code creates the ONNX model, _then_ ORTTrainer(...) attempts to restore state using saved reference.

Reply:
agree with @suffiank , because of the delayed session creation, session is not available when load_state_dict is called. Thus it has to save state_dict_, and strict_ as class members. When later train_step is called with input data so that session is created, it load the check point state into the created session.

Reply:
sorry, I meant to add comment in code explaining a little bit of the motivation...